test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1

```diff
@@ -312,109 +311,115 @@ Describe "Copy-Item can avoid copying an item onto itself" -Tags "CI" {
-
-                    if ($testCase.SelfCopyExpected)
-                    {
-                        # we expected the copy to fail and it did not
-                        throw "Copy-Item should not have succeeded!"
-                    }
-                    else
-                    {
-                        # we expected the copy to succeed. make sure it did
-                        Test-Path $testCase.Destination | Should Be $true
-                        Remove-Item -Path $testCase.Destination -Force -ErrorAction SilentlyContinue
-                    }
-                }
-                catch
-                {
-                    # we expected the copy to fail and it did. make sure it failed the right way
-                    $_.FullyQualifiedErrorId | Should Be "CopyError,Microsoft.PowerShell.Commands.CopyItemCommand"
-                    $_.Exception | Should BeOfType System.IO.IOException
-                    $_.Exception.Data[$selfCopyKey] | Should Not Be $null
-                }
-            }
+    Context "Copy-Item using different case (on case-sensitive file systems)" {
+        BeforeEach {
+            $sourcePath = $filePath
+            $destinationPath = "$TestDrive/" + $fileName.Toupper()
+        }
+        AfterEach {
+            Remove-Item -Path $destinationPath -Force -ErrorAction SilentlyContinue
         }
 
-        # Set up our test cases
-        $testCases = @(
-            @{
-                Name = "Copy to same path"
-                Source = $otherFile
-                Destination = $otherFile
-                SelfCopyExpected = $true
-            }
-            @{
-                Name = "Copy to similar path, different case"
-                Source = $otherFile
-                Destination = "$otherSubDir/" + $otherFileName.ToUpper()
-                SelfCopyExpected = -Not $IsCaseSensitive
-            }
-            @{
-                Name = "Copy hard link"
-                Source = $hardToOtherFile
-                Destination = $otherFile
-                SelfCopyExpected = $true
+        It "Copy-Item can copy to file name differing only by case" {
+            if ($isCaseSensitive)
+            {
+                Copy-Item -Path $sourcePath -Destination $destinationPath -ErrorAction SilentlyContinue | Should Be $null
+                Test-Path -Path $destinationPath | Should Be $true
             }
-            @{
-                Name = "Copy hard link, reversed"
-                Source = $otherFile
-                Destination = $hardToOtherFile
-                SelfCopyExpected = $true
+            else
+            {
+                Copy-Item -Path $sourcePath -Destination $destinationPath -ErrorAction SilentlyContinue | ShouldBeErrorId "CopyError,Microsoft.PowerShell.Commands.CopyItemCommand"
+                $_.Exception | Should BeOfType System.IO.IOException
+                $_.Exception.Data[$selfCopyKey] | Should Not Be $null
```

Comment:
I've changed the test code to use `$Error[0]` rather than `$_`. I'll gladly update to use your template when `ShouldBeErrorId` returns something other than `Boolean`.